controller.xql

```diff
@@ -105,37 +114,70 @@ else if (contains($exist:path, "/public/") and (ends-with($exist:resource, ".png
  : - shared-resources v0.8.4 and earlier (fixed in https://github.com/eXist-db/shared-resources/releases/tag/v0.8.5) --> and thus all versions of eXist up to and including v4.7.0.
  :)
-else if ($exist:path eq "/modules/find.xql") then
+else if ($exist:path eq "/modules/find.xql") then (
+    response:set-status-code(301),
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-        <redirect url="../find?{request:get-query-string()}"/>
+        <redirect url="{$app-root-absolute-url}/find?{request:get-query-string()}"/>
     </dispatch>
-
-else if ($exist:path eq "/find") then
+)
+else if (request:get-method() eq "GET" and $exist:path eq "/find") then
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-        <forward url="modules/find.xq">
+        <forward url="/{$exist:controller}/modules/find.xq">
             <add-parameter name="app-root-absolute-url" value="{$app-root-absolute-url}"/>
         </forward>
     </dispatch>
 
-else if ($exist:resource eq "feed.xml") then
+else if (request:get-method() eq "GET" and $exist:resource eq "feed.xml" or $exist:resource eq "feed" ) then
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-        <forward url="modules/feed.xq"/>
+        <forward url="/{$exist:controller}/modules/feed.xq"/>
     </dispatch>
 
-else if (contains($exist:path, "/$shared/")) then
+else if (request:get-method() eq "GET" and contains($exist:path, "/$shared/")) then
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
         <forward url="/shared-resources/{substring-after($exist:path, '/$shared/')}">
             <set-header name="Cache-Control" value="max-age=3600, must-revalidate"/>
         </forward>
     </dispatch>
 
-else if (contains($exist:path, "/resources/")) then
+else if (request:get-method() eq "GET" and contains($exist:path, "/resources/")) then
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
         <set-header name="Cache-Control" value="max-age=3600, must-revalidate"/>
     </dispatch>
-    
-else
-    (: everything else is passed through :)
+
+(: redirect any unauthorized request to the login page :)
+else if (not(local:is-authorized-user()) and 
+    ($exist:path eq "/admin" or $exist:path eq "/publish")) then
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-        <cache-control cache="yes"/>
+        <forward url="login.html"/>
+        <view>
+            <forward url="{$exist:controller}/modules/view.xq">
+                <set-header name="Cache-Control" value="no-cache"/>
+            </forward>
+        </view>
+    </dispatch>
+
+(: 
+ : Protected resources:
+ : user is required to be logged in and member of a specific group 
+ :)
+else if (request:get-method() = ("GET", "POST") and $exist:path eq "/admin") then
+    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
+        <forward url="admin.html"/>
```

Comment:
Another opportunity for `{$exist:controller}` absolute URL consistency.

Reply:
and that is still here ! :(